crates/matrix-sdk-ui/src/notifications/mod.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for that specific language governing permissions and
+// limitations under the License.
+
+//! Notification API.
+//!
+//! The notification API is a high-level helper that is designed to take care of
+//! handling the synchronization of notifications, be they received within the
+//! app or within a dedicated notification process (e.g. the NSE process on iOS
+//! devices).
+//!
+//! Under the hood, this uses a sliding sync instance configured with no lists,
+//! but that enables the e2ee and to-device extensions, so that it can both
+//! handle encryption et manage encryption keys; that's sufficient to decrypt
+//! messages received in the notification processes.
+//!
+//! As this may be used across different processes, this also makes sure that
+//! there's only one process writing to the databases holding encryption
+//! information. TODO as of 2023-06-06, this hasn't been done yet.
+
+use async_stream::stream;
+use futures_core::stream::Stream;
+use futures_util::{pin_mut, StreamExt};
+use matrix_sdk::{sliding_sync::ExtensionsSetter, Client, SlidingSync};
+use tracing::warn;
+
+/// High-level helper for synchronizing notifications using sliding sync.
+///
+/// See the module's documentation for more details.
+#[derive(Clone)]
+pub struct NotificationApi {
```

Comment:
Just me or is it really weird for `Api` to be part of a type name?

Reply:
Not only you. Please @bnjbvr, can you rename to `Notification` instead?

Reply:
Yep, happy to rename; that being said it's not quite a `Notification`, it's something that handles the notifications for you. The `Manager` suffix always sounds boring and non-descriptive, so not sure what a great name would be here.